### PR TITLE
[TS] [Urgent] LPS-190214

### PR DESCRIPTION
**Pull request by marianoalvarosaiz**

### CODE REVIEW

**Comment by Kvale1733** [comment]
ci:test:users

File: portal-impl/src/com/liferay/portal/service/impl/GroupLocalServiceImpl.java [hunk 1]
```diff
@@ -551,71 +551,82 @@ public Group addGroup(
...
-
-		try {
-			reindexUsersInOrganization(organizationId);
-		}
-		catch (PortalException portalException) {
-			throw new SystemException(portalException);
-		}
+	public boolean addOrganizationGroup(long organizationId, Group group) {
+		return addOrganizationGroup(organizationId, group.getGroupId());
 	}
```
Comment (yuhai) [inline portal-impl/src/com/liferay/portal/service/impl/GroupLocalServiceImpl.java:559]: @marianoalvarosaiz, the commit message LPS number is not right.

File: modules/apps/blogs/blogs-service/src/main/java/com/liferay/blogs/internal/model/listener/UserGroupModelListener.java [hunk 2]
```diff
@@ -58,14 +58,15 @@ private void _unsubscribeUserFromUserGroupGroups(
...
+		List<Group> userGroups = _groupLocalService.getUserGroups(userId, true);
 
-		for (Group group : userGroupGroups) {
-			_blogsEntryLocalService.unsubscribe(userId, group.getGroupId());
+		for (long groupId : userGroupGroupIds) {
+			if (!userGroups.contains(groupId)) {
+				_blogsEntryLocalService.unsubscribe(userId, groupId);
+			}
 		}
 	}
```
Comment (yuhai) [inline modules/apps/blogs/blogs-service/src/main/java/com/liferay/blogs/internal/model/listener/UserGroupModelListener.java:71]: @marianoalvarosaiz, `userGroups.contains(groupId)` is different object. For my previous provided suggestion which is not enough, sorry for this. The following codes should be right.
```
List<Long> groupIds = ListUtil.toList(
			_groupLocalService.getUserGroups(userId, true), Group::getGroupId);

		for (long groupId :
				_userGroupLocalService.getGroupPrimaryKeys(userGroupId)) {

			if (!groupIds.contains(Long.valueOf(groupId))) {
				_blogsEntryLocalService.unsubscribe(userId, groupId);
			}
		}
```

**Comment by yuhai** [comment]
@marianoalvarosaiz, it seems we have different opinions about my review comments https://github.com/liferay-core-infra/liferay-portal/pull/5143#pullrequestreview-1553998401 and https://github.com/liferay-core-infra/liferay-portal/pull/5143#discussion_r1278819123. You didn't do the related changes in this pr. I will ask this with Dante to check if your current changes are good.

**Comment by yuhai** [comment]
@marianoalvarosaiz, after talking this with Dante, we want to use my mentioned way. 

1. If the original logic used **super**, please use **super**. If the original logic used **Persistence**, please use **Persistence**.  
2. If the original logic used model as param, please don't convert it and invoked another method with the param ModelId.

Doing this, the commit will looks clear.

cc @dantewang, @tinatian 

Thanks,
Hai

**Comment by marianoalvarosaiz** [comment]
Hi @yuhai,
 sorry, I didn't see your last message yesterday.

Regarding what you point:

> 
    If the original logic used super, please use super. If the original logic used Persistence, please use Persistence.
    If the original logic used model as param, please don't convert it and invoked another method with the param ModelId.

Which are the benefits besides leaving the commit clearer (which can be solved committing the changes in several commits: one only for the change, one to change persistence to super and one to convert the model)? Because besides that we are duplicating code and having calls in some places to super and in other to persistence which makes the code less maintainable.

Would you consider splitting the commit in several ones?

**Comment by catarineaguiar** [comment]
should QA wait to review?

**Pull request closed without merging**